### ISSUE TITLE
about finetune

### ISSUE THREAD

**Issue by tj-messi** [sbasu276/FocusMAE#4]
Do you finetune the classification task by using the videomae ways?

**Comment by tj-messi** [comment]
by using this:


      class VisionTransformer(nn.Module):
          """ Vision Transformer with support for patch or hybrid CNN input stage
          """
      
          def __init__(self,
                       img_size=224,
                       patch_size=16,
                       in_chans=3,
                       num_classes=1000,
                       embed_dim=768,
                       depth=12,
                       num_heads=12,
                       mlp_ratio=4.,
                       qkv_bias=False,
                       qk_scale=None,
                       drop_rate=0.,
                       attn_drop_rate=0.,
                       drop_path_rate=0.,
                       head_drop_rate=0.,
                       norm_layer=nn.LayerNorm,
                       init_values=0.,
                       use_learnable_pos_emb=False,
                       init_scale=0.,
                       all_frames=16,
                       tubelet_size=2,
                       use_mean_pooling=True,
                       with_cp=False,
                       cos_attn=False):
              super().__init__()
              self.num_classes = num_classes
              # num_features for consistency with other models
              self.num_features = self.embed_dim = embed_dim
              self.tubelet_size = tubelet_size
              self.patch_embed = PatchEmbed(
                  img_size=img_size,
                  patch_size=patch_size,
                  in_chans=in_chans,
                  embed_dim=embed_dim,
                  num_frames=all_frames,
                  tubelet_size=tubelet_size)
              num_patches = self.patch_embed.num_patches
              self.with_cp = with_cp
      
              if use_learnable_pos_emb:
                  self.pos_embed = nn.Parameter(
                      torch.zeros(1, num_patches, embed_dim))
              else:
                  # sine-cosine positional embeddings is on the way
                  self.pos_embed = get_sinusoid_encoding_table(
                      num_patches, embed_dim)
      
              self.pos_drop = nn.Dropout(p=drop_rate)
      
              dpr = [x.item() for x in torch.linspace(0, drop_path_rate, depth)
                     ]  # stochastic depth decay rule
              self.blocks = nn.ModuleList([
                  Block(
                      dim=embed_dim,
                      num_heads=num_heads,
                      mlp_ratio=mlp_ratio,
                      qkv_bias=qkv_bias,
                      qk_scale=qk_scale,
                      drop=drop_rate,
                      attn_drop=attn_drop_rate,
                      drop_path=dpr[i],
                      norm_layer=norm_layer,
                      init_values=init_values,
                      cos_attn=cos_attn) for i in range(depth)
              ])
              self.norm = nn.Identity() if use_mean_pooling else norm_layer(
                  embed_dim)
              self.fc_norm = norm_layer(embed_dim) if use_mean_pooling else None
              self.head_dropout = nn.Dropout(head_drop_rate)
              self.head = nn.Linear(
                  embed_dim, num_classes) if num_classes > 0 else nn.Identity()
      
              if use_learnable_pos_emb:
                  trunc_normal_(self.pos_embed, std=.02)
      
              self.apply(self._init_weights)
      
              self.head.weight.data.mul_(init_scale)
              self.head.bias.data.mul_(init_scale)
      
          def _init_weights(self, m):
              if isinstance(m, nn.Linear):
                  trunc_normal_(m.weight, std=.02)
                  if isinstance(m, nn.Linear) and m.bias is not None:
                      nn.init.constant_(m.bias, 0)
              elif isinstance(m, nn.LayerNorm):
                  nn.init.constant_(m.bias, 0)
                  nn.init.constant_(m.weight, 1.0)
      
          def get_num_layers(self):
              return len(self.blocks)
      
          @torch.jit.ignore
          def no_weight_decay(self):
              return {'pos_embed', 'cls_token'}
      
          def get_classifier(self):
              return self.head
      
          def reset_classifier(self, num_classes, global_pool=''):
              self.num_classes = num_classes
              self.head = nn.Linear(
                  self.embed_dim, num_classes) if num_classes > 0 else nn.Identity()
      
          def forward_features(self, x):
              B = x.size(0)
      
              x = self.patch_embed(x)
      
              if self.pos_embed is not None:
                  x = x + self.pos_embed.expand(B, -1, -1).type_as(x).to(
                      x.device).clone().detach()
              x = self.pos_drop(x)
      
              for blk in self.blocks:
                  if self.with_cp:
                      x = cp.checkpoint(blk, x)
                  else:
                      x = blk(x)
      
              if self.fc_norm is not None:
                  return self.fc_norm(x.mean(1))
              else:
                  return self.norm(x[:, 0])
      
          def forward(self, x):
              x = self.forward_features(x)
              x = self.head_dropout(x)
              x = self.head(x)
              return x
